### PR TITLE
provide initial generic AccountsNew Menu, re #3666

### PR DESCRIPTION
**Pull request by ghost**
can this go into a new `feature/multi-x` branch?

(couldn't find a way to add this)

### CODE REVIEW

**Comment by lazaridiscom** [comment]
@kumavis , @danfinlay , how do you manage this scenario here:

I changed MM, and I change an npm-package (KeyringController). The CI does not know about the change in KeyringController.

So naturally, the tests will fail (here).

You know of any standard-solution to this?

File: ui/app/components/account-menu/index.js [hunk 1]
```diff
@@ -103,6 +119,53 @@ AccountMenu.prototype.render = function () {
...
+    keyringProviders,
+    showNewAccountPage, // needed thus onClick event fires properly
+  } = this.props
+
+  var temp=[];
+  Object.keys(keyringProviders).map((key, index) => {
+
+    const func = keyringProviders[key].func;
+    const text = keyringProviders[key].text;
+    const img  = keyringProviders[key].img;
```
Comment (danfinlay) [inline ui/app/components/account-menu/index.js]: You could do this in one line using destructuring (very fancy!):
`const { func, text, img } = keyringProviders[key]`

**Review by danfinlay — COMMENTED** [review]
Small JS tip.

**Comment by danfinlay** [comment]
I've now created a branch off `master` called `feature/multi-x`, so you can submit pull requests to that, including periodically merging `master` into it, if this is sustainable.

For developing against your own fork of a published module, I would put your module in the `package.json` and `package-lock.json`, like this:

```
npm install lazaridiscom/eth-keyring-controller#feature/multi-x -S
```

You may need to re-do this periodically, since the `package-lock.json` will hash-lock the dependency against a certain version.

While developing locally, there are a couple strategies you can use to allow live-reloading the project when changing a local dependency. One is this:
https://github.com/MetaMask/metamask-extension/blob/master/docs/developing-on-deps.md

Closing for now, because we don't want to merge this into `master`.

**Comment by lazaridiscom** [comment]
> For developing against your own fork of a published module

its not a published module (just a git repo: https://github.com/lazaridiscom/mm-vault). So I possibly will check the "submodules" feature of git. Possibly the ci honors submodules already by default.

> strategies you can use to allow live-reloading

@danfinlay , strangely, live reload works for me, without any of the steps described in the doc you pointed to.

**Pull request closed without merging**